### DISCUSSION TITLE
mq/100-budiushi

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#169] [Announcements]
# mq/100-budiushi

MQ：怎么确保消息100%不丢失？ 面试官在面试候选人时，如果发现候选人的简历中写了在项目中使用了 MQ 技术（如 Kafka、RabbitMQ、RocketMQ），基本都会抛出一个问题：在使用 MQ 的时候，怎么确保消息 100% 不丢失？ 这个问题在实际工作中很常见，既能考察候选者对于 MQ 中间件技术的掌握程度，又能很好地区分候选人的能力水平。接下来，我们就从这个问题出发，探讨你应该掌握的基础知识和答题思路，以及延伸的面试考点。 案例背景 以京东系统为例，用户在购买商品时，通常会选择用京豆抵扣一部分的金额，在这个过程中，交易服务和京豆服务通过 MQ 消息队列进行通信。在下单时，交易服务发送“扣减账户 X 100 个京豆”的消息给 MQ 消息队列，而京豆服务则在消费端消费这条命令，实现真正的扣减操作。

https://tobebetterjavaer.com/mq/100-budiushi.html

**Comment by YiNianR** [comment]
讲的很清晰
